Library/Homebrew/cmd/update.sh

```diff
@@ -640,10 +631,31 @@ EOS
 
       # HOMEBREW_UPDATE_FORCE and HOMEBREW_UPDATE_AUTO aren't modified here so ignore subshell warning.
       # shellcheck disable=SC2030
-      if [[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]]
+      if [[ "${UPSTREAM_REPOSITORY_URL}" = "https://github.com/"* ]] ||
+         [[ "${UPSTREAM_REPOSITORY_URL}" =~ https://(([^:@]+)(:([^@]+))?@)?github.com/(.*)$ ]]
       then
-        UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY_URL#https://github.com/}"
-        UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY%.git}"
+        if [[ -n "${BASH_REMATCH[5]}" ]]
+        then
+          UPSTREAM_REPOSITORY="${BASH_REMATCH[5]%.git}"
+        else
+          UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY_URL#https://github.com/}"
+          UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY%.git}"
+        fi
+        # HOMEBREW_GITHUB_API_TOKEN is optionally defined in the user environment.
+        # Global token HOMEBREW_GITHUB_API_TOKEN supersedes local defined one
+        # shellcheck disable=SC2153
+        if [[ -n "${HOMEBREW_GITHUB_API_TOKEN}" ]]
```

Comment:
```suggestion
        if [[ "${UPSTREAM_REPOSITORY_URL}" = "https://github.com/"* ]] && 
           [[ -n "${HOMEBREW_GITHUB_API_TOKEN}" ]]
```

or similar as otherwise `HOMEBREW_GITHUB_API_TOKEN` being set at all may limit this.

It may be desirable to have a single `[[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]]` check which sets a variable e.g. `UPSTREAM_REPOSITORY_GITHUB_GLOB` or something which you can then check in these multiple places later.